sdk/core/azure-core/src/main/java/com/azure/core/util/FluxUtil.java

```diff
@@ -65,6 +65,46 @@ public static boolean isFluxByteBuffer(Type entityType) {
         return false;
     }
 
+    /**
+     * Adds progress reporting to the provided {@link Flux} of {@link ByteBuffer}.
+     *
+     * <p>
+     *     Each {@link ByteBuffer} that's emitted from the {@link Flux} will report {@link ByteBuffer#remaining()}.
+     * </p>
+     * <p>
+     *     When {@link Flux} is resubscribed the progress is reset.
```

Comment:
This assumes that the flux is replayable. If the flux is not replayable, resubscribing can result in empty or partial data and the progress reporter might not be accurate. We should document this behavior.